lib/cog/models/ecto_json.ex

```diff
@@ -52,6 +52,37 @@ defmodule Cog.Models.EctoJson do
     end
   end
 
+  def render_permissions(model, opts \\ []) do
+    opts = Keyword.merge(@render_defaults, opts)
+
+    policy = Keyword.fetch!(opts, :policy)
+    data = do_render(model, policy)
+    |> format_permissions(model)
+
+    case Keyword.fetch!(opts, :envelope) do
+      nil ->
+        data
+      envelope ->
+        Map.put(%{}, envelope, data)
+    end
+  end
+
+  def format_permissions(data, []), do: data
+  def format_permissions(data, %{__struct__: Cog.Models.Role}), do: data
+  def format_permissions(data, model) when is_list(model) do
+    [item | _] = model
+    format_permissions(data, item)
+  end
+  def format_permissions(data, _model) when is_list(data) do
+    perms = Enum.group_by(data, &(Map.get(&1, :namespace)))
+    Enum.into(Map.keys(perms), %{}, &scrub(&1, perms, :namespace))
+  end
+  def format_permissions(data, _model) when is_map(data) do
+    {ns, scrubbed} = Map.pop(data, :namespace)
+    %{ns => [scrubbed]}
+  end
+  def format_permissions(data, _model), do: data
```

Comment:
I kinda thing `EctoJson` might be the wrong place for this, but I don't really have a good idea of where it would go. If we were using phoenix views a `permission` view would be the perfect place to put this kind of stuff. I wonder why we're not using those, hrm.